wordpress-mobile/WordPress-Android
# Jetpack App (Basics): Site Address is not Jetpack

Parent #14302 

This PR implements the empty state UI when using a site address to log into the Jetpack app and it's not a Jetpack site or it's disconnected.

Implementation Details
- Introduced `LoginSiteCheckErrorViewModel` to handle button clicks and back press
- Refactored the way binding was being handled in `LoginSiteCheckErrorFragment`
- Refactored `ShowInstructions` navigation event to use a single const val for the URL, but still allow flexibility if this changes in the future by passing the value as a param

**Merge Instructions**
- Ensure PR #14596 has been merged
- Remove Not ready for merge label
- Merge as normal

**To test:**
- Download this branch and build a Jetpack App variant
- Tap the 'Enter your existing address' button
- Enter `http://do.wpmt.co/jp-disconnected/` or another site that has Jetpack installed but not connected
- You should be brought to an empty view with the message "The website {site} is not a Jetpack site"

## Regression Notes
1. Potential unintended areas of impact 🟢 

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual and automated tests

3. What automated tests I added (or what prevented me from doing so)
Added new unit tests in `LoginSiteCheckErrorViewModelTest`

PR submission checklist:
